src/components/Chart/ChartContext.js

```diff
@@ -0,0 +1,215 @@
+    : d => d.category
+
+  const colorValues = []
+    .concat(data.map(colorAccessor))
+    .concat(mergedProps.colorLegendValues)
+    .filter(Boolean)
+    .filter(deduplicate)
+
+  const xValuesUnformatted = data
+    .map(xAccessor)
+    .concat(props.xTicks ? props.xTicks.map(xNormalizer) : [])
+    .concat(getAnnotationsXValues(xAnnotations, xNormalizer))
+    .filter(deduplicate)
+
+  const processedData = dataProcesser[type]({
+    props: mergedProps,
+    data,
+    colorAccessor,
+    colorValues,
+    xValuesUnformatted,
+    xFormat,
+    xParser,
+    xParserFormat,
+    xSort,
+    xNormalizer
+  })
+
+  const chartContextObject = {
+    ...processedData,
+    colorAccessor,
+    colorValues,
+    xNormalizer
+  }
+
+  return (
+    <ChartContext.Provider value={chartContextObject}>
+      {mergedProps.children}
+    </ChartContext.Provider>
+  )
+}
+
+export const defaultProps = {
+  TimeBar: {
+    x: 'year',
+    xScale: 'time',
+    xBandPadding: 0.25,
+    timeParse: '%Y',
+    numberFormat: 's',
+    height: 240,
+    padding: 50,
+    unit: '',
+    xUnit: '',
+    colorLegend: true,
+    xIntervalStep: 1,
+    yAnnotations: [],
+    xAnnotations: [],
+    columns: 1,
+    minInnerWidth: 240
+  },
+  Line: {
+    x: 'year',
+    xScale: 'time',
+    yScale: 'linear',
+    timeParse: '%Y',
+    timeFormat: '%Y',
+    numberFormat: '.0%',
+    zero: true,
+    unit: '',
+    startValue: false,
+    endValue: true,
+    endLabel: true,
+    endDy: '0.3em',
+    minInnerWidth: 110,
+    columns: 1,
+    height: 240,
+    yNice: 3
+  },
+  Slope: {
+    x: 'year',
+    xScale: 'ordinal',
+    yScale: 'linear',
+    timeParse: '%Y',
+    timeFormat: '%Y',
+    numberFormat: '.0%',
+    zero: true,
+    unit: '',
+    startValue: true,
+    endValue: true,
+    endLabel: false,
+    endDy: '0.3em',
+    minInnerWidth: 90,
+    columns: 2,
+    height: 240,
+    yNice: 3
+  }
+}
+
+const axisPropType = PropTypes.shape({
+  ticks: PropTypes.array.isRequired,
+  format: PropTypes.func.isRequired,
+  axisFormat: PropTypes.func.isRequired
+})
+
+const propTypes = {
+  groupedData: PropTypes.array.isRequired,
+  groupPosition: PropTypes.shape({
+    y: PropTypes.func.isRequired,
+    x: PropTypes.func.isRequired,
+    titleHeight: PropTypes.number // timebar only
+  }).isRequired,
+  height: PropTypes.number.isRequired,
+  innerWidth: PropTypes.number.isRequired,
+  x: PropTypes.func.isRequired,
+  xAxis: axisPropType.isRequired,
+  xDomain: PropTypes.array.isRequired,
+  y: PropTypes.func.isRequired,
+  // inconsistent
+  colorValues: PropTypes.array,
+  colorLegend: PropTypes.bool,
+  colorLegendValues: PropTypes.array,
+  // only used by timebar
+  colorAccessor: PropTypes.func,
+  xNormalizer: PropTypes.func,
+  // timebar only
+  xValues: PropTypes.array,
+  // line only
+  paddingLeft: PropTypes.number,
+  paddingRight: PropTypes.number,
+  columnHeight: PropTypes.number,
+  yAxis: axisPropType,
```

Comment:
this should probably eventually become `isRequired` and also be provided by the time bar context. but can wait for another PR